mantidproject/mantid
# Indirect - ConvFit - Q value not updating when spectrum changed

**Description of work.**
This PR ensures that the Q value and WorkspaceIndex in ConvFit is updated when the spectrum has been changed using the `PlotSpectrum` spinbox.

**To test:**
1.`Interfaces`->`Indirect`->`Data Analysis`->`ConvFit` tab
2. Load the data below
3. FitType is `TeixeiraWater`
4. Background is `FlatBackground`
4. Look for Q value and workspace Index in the FitPropertyBrowser
5. Change the spectrum selected using the `Plot Spectrum` spinbox
6. The Q value and WorkspaceIndex should update!
7. Change the WorkspaceIndex using the FitPropertyBrowser, the `PlotSpectrum` spinbox number should change.

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2561154/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2561153/irs26173_graphite002_res.zip)

Fixes #23950 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
